realm/realm-library/src/androidTest/java/io/realm/ManagedOrderedRealmCollectionTests.java

```diff
@@ -0,0 +1,749 @@
+                realm.commitTransaction();
+                return obj.getFieldListNull();
+
+            case REALMRESULTS:
+                return realm.where(NullTypes.class).findAll();
+        }
+
+        throw new AssertionError("Unknown collection: " + collectionClass);
+    }
+
+    @After
+    public void tearDown() {
+        realm.close();
+    }
+
+    @Test
+    public void sort_twoFields() {
+        OrderedRealmCollection<AllJavaTypes> sortedList = collection.sort(AllJavaTypes.FIELD_BOOLEAN, Sort.ASCENDING, AllJavaTypes.FIELD_LONG, Sort.DESCENDING);
+        AllJavaTypes obj = sortedList.first();
+        assertFalse(obj.isFieldBoolean());
+        assertEquals(TEST_SIZE - 1, obj.getFieldLong());
+    }
+
+    @Test
+    public void sort_boolean() {
+        OrderedRealmCollection<AllJavaTypes> sortedList = collection.sort(AllJavaTypes.FIELD_BOOLEAN, Sort.DESCENDING);
+        assertEquals(TEST_SIZE, sortedList.size());
+        assertEquals(false, sortedList.last().isFieldBoolean());
+        assertEquals(true, sortedList.first().isFieldBoolean());
+        assertEquals(true, sortedList.get(TEST_DATA_FIRST_HALF).isFieldBoolean());
+        assertEquals(false, sortedList.get(TEST_DATA_LAST_HALF).isFieldBoolean());
+
+        RealmResults<AllJavaTypes> reverseList = sortedList.sort(AllJavaTypes.FIELD_BOOLEAN, Sort.ASCENDING);
+        assertEquals(TEST_SIZE, reverseList.size());
+        assertEquals(true, reverseList.last().isFieldBoolean());
+        assertEquals(false, reverseList.first().isFieldBoolean());
+        assertEquals(false, reverseList.get(TEST_DATA_FIRST_HALF).isFieldBoolean());
+        assertEquals(true, reverseList.get(TEST_DATA_LAST_HALF).isFieldBoolean());
+
+        RealmResults<AllJavaTypes> reserveSortedList = reverseList.sort(AllJavaTypes.FIELD_BOOLEAN, Sort.DESCENDING);
+        assertEquals(TEST_SIZE, reserveSortedList.size());
+        assertEquals(reserveSortedList.first(), sortedList.first());
+    }
+
+    @Test
+    public void sort_string() {
+        OrderedRealmCollection<AllJavaTypes> resultList = collection;
+        OrderedRealmCollection<AllJavaTypes> sortedList = createCollection(collectionClass);
+        sortedList = sortedList.sort(AllJavaTypes.FIELD_STRING, Sort.DESCENDING);
+
+        assertEquals(resultList.size(), sortedList.size());
+        assertEquals(TEST_SIZE, sortedList.size());
+        assertEquals(resultList.first().getFieldString(), sortedList.last().getFieldString());
+
+        RealmResults<AllJavaTypes> reverseList = sortedList.sort(AllJavaTypes.FIELD_STRING, Sort.ASCENDING);
+        assertEquals(TEST_SIZE, reverseList.size());
+        assertEquals(resultList.first().getFieldString(), reverseList.first().getFieldString());
+
+        int numberOfDigits = 1 + ((int) Math.log10(TEST_SIZE));
+        int largestNumber = 1;
+        for (int i = 1; i < numberOfDigits; i++)
+            largestNumber *= 10;  // 10*10* ... *10
```

Comment:
can't you use [Math.pow](http://developer.android.com/reference/java/lang/Math.html#pow(double, double)) ?